src/logic/stopPlaceMatching/index.ts

```diff
@@ -0,0 +1,197 @@
+
+// searches starting within 30m of the user's current position
+let shouldHaveFound = 0
+
+// closest stopPlace and within 30m
+let correctlyFound = 0
+
+// real first stop place is within maxDistanceMeters of GPS position but wasn't found as any nearby stop
+let missedCompletely = 0
+
+// real first stop place is > maxDistanceMeters away
+let tooFarAway = 0
+
+// remember that 0 here is the number of cases with only 1 nearby stop etc
+const nearbyCount: number[] = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+
+// ideally we want to find everything at index 0.
+const foundAtIndex: number[] = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+
+// distance to correct stop place in groups of 10m (0-10, 10-20 etc)
+const foundAtDistance: number[] = [0, 0, 0, 0, 0, 0]
+
+const startTime = Date.now()
+let lastLogTime = Date.now()
+
+// This is a fragile way of detecting that the user has searched for 'your position' but currently it
+// is the only way of knowing.
+const myLocationTexts = ['Posisjonen din', 'Your location']
+
+const logNearbyStatistics = (): void => {
+    if (Date.now() - lastLogTime > LOG_INTERVAL) {
+        lastLogTime = Date.now()
+        const hitRate =
+            Math.round((1000 * correctlyFound) / shouldHaveFound) / 10
+        logger.info(
+            `(id: ${INSTANCE_ID}) Nearby-matching: Hit rate ${hitRate}. Expand for statistics`,
+            {
+                instanceId: INSTANCE_ID,
+                runningTime: `${Math.round((lastLogTime - startTime) / 1000)}s`,
+                totalChecks,
+                hitRate: {
+                    hitRate,
+                    shouldHaveFound,
+                    correctlyFound,
+                    missedCompletely, // missed even when looking at +20m
+                },
+                tooFarAway,
+                foundAtIndex,
+                foundAtDistance,
+                stopPlacesWithoutCoordinates,
+                distancesFromUserGPS: [
+                    ` 0 - 10m: ${foundAtDistance[0]}`,
+                    `10 - 20m: ${foundAtDistance[1]}`,
+                    `20 - 30m: ${foundAtDistance[2]}`,
+                    `30 - 40m: ${foundAtDistance[3]}`,
+                    `40 - 50m: ${foundAtDistance[4]}`,
+                    `50 - 60m: ${foundAtDistance[5]}`,
+                ],
+            },
+        )
+    }
+}
+
+const getDistanceInMeters = (
+    fromLat: number,
+    fromLon: number,
+    toLat: number,
+    toLon: number,
+): number => {
+    const start = { latitude: fromLat, longitude: fromLon }
+    const end = { latitude: toLat, longitude: toLon }
+    return haversine(start, end, { unit: 'meter' })
+}
+
+export const runStopPlaceMatching = async (
+    clientPlatform: string,
+    params: SearchParams,
+    tripPatterns: TripPatternParsed[],
+): Promise<void> => {
+    try {
+        const lon = params.from.coordinates?.longitude
+        const lat = params.from.coordinates?.latitude
+        const isMyLocation = myLocationTexts.includes(params.from.name || '')
```

Comment:
Trenger kanskje ikke søke igjennom arrayet om man ikke har et param.
```suggestion
        const isMyLocation = Boolean(params.from.name && myLocationTexts.includes(params.from.name))
```